File: cpp-test-files/16.cpp
//Konstans
#include <iostream>
#include <string>

class Day {
private:
    const std::string name;
    int temperature;

public:
    Day(const std::string& dayName, int temp)
        : name(dayName), temperature(temp) {}

    std::string getName() const {
        return name;
    }

    int getTemperature() const {
        return temperature;
    }

    void setTemperature(int newTemp) {
        temperature = newTemp;
    }

    void printDayInfo(const std::string& prefix) const {
        std::cout << prefix << " " << name << " has " << temperature << " degrees.\n";
    }
};

File: java-files-corrected/16.java
class Day {
    private String name;
    private int temperature;

    public Day(String dayName, int temp) {
        this.name = dayName + " " + temperature;
        this.temperature = temp;
    }

    public String getName() {
        return name;
    }

    public int getTemperature() {
        return temperature;
    }

    public void setTemperature(int newTemp) {
        temperature = newTemp;
    }

    public void printDayInfo(String prefix) {
        System.out.println(prefix + " " + name + " has " + temperature + " degrees.");
    }
}